R/set_compartment_scale.R

```diff
@@ -4,19 +4,40 @@
 #' Currently not available in Pharmpy, this is a workaround function.
 #'
 #' @inheritParams run_nlme
-#' @param compartment compartment number
+#' @param compartment compartment number. If `NULL` will be attempted to
+#' infer from ADVAN. If not a default ADVAN is used, will use 1 as default. So
+#' for safe use, please always specify the observation compartment to be scaled.
 #' @param expression specification of new scaling, should always contain variable
 #' and scale arguments. E.g. `list(variable = "V", "scale" = 1000)`.
+#' @param update_inits update initial estimates for basic PK parameters? This is
+#' likely needed when applying scale, or else it is very likely that the model
+#' starts too far off from the maximum likelihood and the fit will not
+#' converge properly. `TRUE` by default.
 #'
 #' @returns Pharmpy model
 #'
 #' @export
 set_compartment_scale <- function(
     model,
-    compartment = 1,
+    compartment = NULL,
     expression = list(variable = "V", scale = 1000),
+    update_inits = TRUE,
     verbose = TRUE
 ) {
+  ## if compartment not specified, then try to guess
```

Comment:
this part is unrelated, just QoL addition: it will infer the observation compartment to scale, from the ADVAN number of the model. So for the standard PK models the `compartment` argument can be NULL.